paperqa/types.py

```diff
@@ -593,7 +606,9 @@ def populate_bibtex_key_citation(  # noqa: PLR0912
 
     @model_validator(mode="before")
     @classmethod
-    def validate_all_fields(cls, data: dict[str, Any]) -> dict[str, Any]:
+    def validate_all_fields(cls, data: dict[str, Any] | Doc) -> dict[str, Any]:
+        if isinstance(data, Doc):
+            data = data.model_dump()
```

Comment:
Do you mind reverting this change? I just debugged the test suite, this `Doc` entrance doesn't actually happen in practice